From 40946990cd9d55dd43fa2696b011cc73fe7d2ccc Mon Sep 17 00:00:00 2001
From: Markus Valentin <valy@systemausfall.org>
Date: Tue, 20 Jun 2017 19:04:31 +0200
Subject: [PATCH 5/7] arm: dts: urwerk: reconfigured the (debug)-uarts

Now we have to possible debug uart(s) the external and the one on the
testpads, this allows the following configurations:

    Normal boot mode: DUART on testpads, AUART4 as external serial port
    USB-Debug-Mode:  DUART on external serial port, UART4 disabled

The now configured default is the DUART on the testpads.

Signed-off-by: Markus Valentin <valy@systemausfall.org>
---
 arch/arm/boot/dts/imx28.dtsi | 11 +++++++++++
 arch/arm/boot/dts/urwerk.dts | 30 +++++++++++++++++++++++++++++-
 2 files changed, 40 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx28.dtsi b/arch/arm/boot/dts/imx28.dtsi
index b54dd03f3481..517e2be6215a 100644
--- a/arch/arm/boot/dts/imx28.dtsi
+++ b/arch/arm/boot/dts/imx28.dtsi
@@ -405,6 +405,17 @@
 					fsl,pull-up = <MXS_PULL_DISABLE>;
 				};
 
+				auart4_2pins_b: auart4@1 {
+					reg = <1>;
+					fsl,pinmux-ids = <
+						MX28_PAD_AUART0_RTS__AUART4_TX
+						MX28_PAD_AUART0_CTS__AUART4_RX
+						>;
+					fsl,drive-strength = <MXS_DRIVE_4mA>;
+					fsl,voltage = <MXS_VOLTAGE_HIGH>;
+					fsl,pull-up = <MXS_PULL_DISABLE>;
+				};
+
 				mac0_pins_a: mac0@0 {
 					reg = <0>;
 					fsl,pinmux-ids = <
diff --git a/arch/arm/boot/dts/urwerk.dts b/arch/arm/boot/dts/urwerk.dts
index 68037ce20f0d..b273eabb1f63 100755
--- a/arch/arm/boot/dts/urwerk.dts
+++ b/arch/arm/boot/dts/urwerk.dts
@@ -128,16 +128,38 @@
 				};
 			};
 
+			//*************************************************************
+			// ATTENTION: Only choose one DUART config each time,
+			//            and be careful with AUART4, because they
+			//            are using the same pins.
+			// Intention: Normal boot mode: DUART on testpads, AUART4 as external serial port
+			//            USB-Debug-Mode:  DUART on external serial port, AUART4 disabled
+			//*************************************************************/
+			// duart on Urwerk testpads
 			duart: serial@80074000 {
 				pinctrl-names = "default";
 				pinctrl-0 = <&duart_pins_a>;
 				status = "okay";
 			};
 
+			// duart on Urwerk's external RS232-Pins
+			//duart: serial@80074000 {
+			//	pinctrl-names = "default";
+			//	pinctrl-0 = <&duart_pins_b>;
+			//	status = "disabled";
+			//};
+
+			auart4: serial@80072000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&auart4_2pins_b>;
+				status = "okay";
+			};
+
+			// not used since Urwerk v1.5.3
 			auart0: serial@8006a000 {
 				pinctrl-names = "default";
 				pinctrl-0 = <&auart0_2pins_a>;
-				status = "okay";
+				status = "disabled";
 			};
 
 			auart1: serial@8006c000 {
@@ -146,6 +168,12 @@
 				status = "okay";
 			};
 
+			auart3: serial@80070000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&auart3_2pins_b>;
+				status = "okay";
+			};
+
 			usbphy0: usbphy@8007c000 {
 				status = "okay";
 			};
-- 
2.13.1

